### DISCUSSION TITLE
TerriaJS connection to Postgres/PostGIS

### DISCUSSION THREAD

**Discussion by jdclarke5** [TerriaJS/terriajs#6856] [Q&A]
We have data in a PostgreSQL table, with some columns of geo-type supported by PostGIS. Can we connect these data to TerriaJS? How would we go about doing that. Is there any documentation on this?

**Comment by techmavengeospatial** [comment]
Use dynamic pbf Vector tiles from postgis

Use crunchydata pg_tileserv
Pg_featureserv

**Comment by nf-s** [comment]
Hi @jdclarke5 

TerriaJS is not able to directly connect to PostgreSQL, you would need to setup some sort of web-service in front of your DB.
I am not familiar with @techmavengeospatial's suggestions, but they look suitable!

TerriaJS supports a wide range of data formats and APIs, so there are a lot of potential options.

For example, you could use GeoServer with a PostGIS store - see https://docs.geoserver.org/latest/en/user/data/database/postgis.html - and you can then publish WMS layers (rendered image tiles), WFS layers (vector features) or mapbox vector tiles - which are all supported by TerriaJS.

Let me know if you'd like any more information about what I have mentioned here - and I can also provide examples of WMS, WFS and mapbox vector tile layers in TerriaJS

Thanks 🙂 
Nick
